Source/WebKit/UIProcess/RemoteLayerTree/RemoteScrollingCoordinatorProxy.h

```diff
@@ -129,6 +133,9 @@ class RemoteScrollingCoordinatorProxy {
     std::optional<unsigned> m_currentVerticalSnapPointIndex;
     bool m_propagatesMainFrameScrolls { true };
     HashSet<WebCore::GraphicsLayer::PlatformLayerID> m_layersWithScrollingRelations;
+#if ENABLE(OVERLAY_REGIONS_IN_EVENT_REGION)
+    HashSet<WebCore::GraphicsLayer::PlatformLayerID> m_scrollingNodeFixedLayerIDs;
```

Comment:
Nothing removes entries from this HashSet when layers are destroyed, so it will grow larger and larger. We do send the list of removed layers to the UI process, so you could remove them using that.

Reply:
What if I did something like this?
```
+++ b/Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteScrollingCoordinatorProxyIOS.mm
@@ -75,20 +75,19 @@ UIScrollView *RemoteScrollingCoordinatorProxy::scrollViewForScrollingNodeID(Scro
 void RemoteScrollingCoordinatorProxy::connectStateNodeLayers(ScrollingStateTree& stateTree, const RemoteLayerTreeHost& layerTreeHost)
 {
     using PlatformLayerID = GraphicsLayer::PlatformLayerID;
+#if ENABLE(OVERLAY_REGIONS_IN_EVENT_REGION)
+    m_scrollingNodeFixedLayerIDs.clear();
+#endif
 
     for (auto& currNode : stateTree.nodeMap().values()) {
-        if (currNode->hasChangedProperty(ScrollingStateNode::Property::Layer)) {
-            auto platformLayerID = PlatformLayerID { currNode->layer() };
+        auto platformLayerID = PlatformLayerID { currNode->layer() };
+        if (currNode->hasChangedProperty(ScrollingStateNode::Property::Layer))
             currNode->setLayer(layerTreeHost.layerForID(platformLayerID));
+
 #if ENABLE(OVERLAY_REGIONS_IN_EVENT_REGION)
-            if (platformLayerID) {
-                if (currNode->isFixedNode())
-                    m_scrollingNodeFixedLayerIDs.add(platformLayerID);
-                else
-                    m_scrollingNodeFixedLayerIDs.remove(platformLayerID);
-            }
+        if (platformLayerID && currNode->isFixedNode())
+            m_scrollingNodeFixedLayerIDs.add(platformLayerID);
 #endif
-        }
```

which clears the set and rebuilds it every time `connectStateNodeLayers` is called?

Reply:
Are those tracked by `RemoteLayerTreeTransaction::destroyedLayers()` or should I be getting the destroyed list from elsewhere? I tried iterating over that list in `_didCommitLayerTree` but the list of destroyed layer IDs doesn't seem to match up with the scrolling ID nodes so they weren't getting removed. In an update I will propose checking the layer which resolves the issue but I'm not sure if we are concerned of the inefficiencies given the number of fixed scrolling nodes. However I will limit it to when there are layer tree updates which maybe is less frequent than every time `RemoteScrollingCoordinatorProxy::connectStateNodeLayers` is called?